### PR TITLE
Remove `redirect_uri` support from OIDC logout endpoint

### LINKED ISSUE #10983: Remove redirect_uri support from OIDC logout endpoint
### Description

Keycloak 18 was updated to follow the RP initiated logout specification, deprecating the support for the legacy `redirect_uri` parameter on the logout endpoint. Support for this backwards compatibility mode should be removed at some point.

### Details

Removing this will require:

- Update of the `LogoutEndpoint.logout()` method and remove support for "redirect_uri" parameter of this method
- Removing method `RedirectUtils.verifyRealmRedirectUri(KeycloakSession session, String redirectUri)` (which is already deprecated in Keycloak 18). And remove some other private method in the `RedirectUtils` class, which is used by the removed `verifyRealmRedirectUri` .
- Remove configuration option `OIDCLoginProtocolFactory.CONFIG_LEGACY_LOGOUT_REDIRECT_URI` and `OIDCLoginProtocolFactory.SUPPRESS_LOGOUT_CONFIRMATION_SCREEN` from `OIDCLoginProtocolFactory` and this constant. This also means reverting changes introduced by https://github.com/keycloak/keycloak/pull/13471 . Probably whole class `OIDCProviderConfig` can be removed in this case (unless we have some other configuration option to `OIDCLoginProtocolFactory` added in the meantime)
- Remove LegacyLogoutTest
- Remove system property config option from the keycloak-server.json in the testsuite
- Documentation will need to be updated (Upgrading guide) about the fact that the options, which were deprecated in Keycloak 18 and 19.0.2, are removed now.

Will be good to discuss with @sschu and/or @bs-matil before removing this.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #10983

### CODE REVIEW

File: services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java [hunk 1]
```diff
@@ -215,8 +213,9 @@ public Object issueUserInfo() {
     /* old deprecated logout endpoint needs to be removed in the future
     * https://issues.redhat.com/browse/KEYCLOAK-2940 */
     @Path("logout")
+    @Deprecated
     public Object logout() {
```
Comment (jonkoops) [inline services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java:217]: Looks like this logout endpoint has been deprecated for some time, do we have an issue to track it's removal?

Comment (sschu) [reply services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java:217]: I am not 100% sure but I think the comment does not target the whole endpoint but just the functionality within the endpoint that is deprecated, see e.g. https://github.com/keycloak/keycloak/blob/a65508ca13da5ad41ae0f4aa045ae0dd6219f3dc/services/src/main/java/org/keycloak/protocol/oidc/endpoints/LogoutEndpoint.java#L165-L185

**Review by sschu — COMMENTED** [review]
In general I am fine with removing the legacy support, my only point would be the aforementioned marking of the endpoint as deprecated.

File: services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java (hunk 1 above)
Comment (jonkoops) [reply services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java:214]: You're right, I've removed the `@Deprecated` annotation and the comment.

**Review by sschu — APPROVED** [review]
LGTM.

**Review by abstractj — DISMISSED** [review]

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc [hunk 2]
```diff
@@ -183,3 +183,15 @@ Additionally, the following resources have been removed from the `common` theme:
...
+
+Previous versions of {project_name} had supported automatic logout of the user and redirecting to the application by opening logout endpoint URL such as
+`http(s)://example-host/auth/realms/my-realm-name/protocol/openid-connect/logout?redirect_uri=encodedRedirectUri`. This functionality was deprecated in {project_name} 18 and has been removed in this version in favor of following the OpenID Connect specification.
+
+As part of this change the following related configuration options for the SPI have been removed:
+
+- `--spi-login-protocol-openid-connect-legacy-logout-redirect-uri`
+- `--spi-login-protocol-openid-connect-suppress-logout-confirmation-screen`
+
+If you were still making use these options or the `redirect_uri` paramater for logout you should implement the link:|https://openid.net/specs/openid-connect-rpinitiated-1_0.html[OpenID Connect RP-Initiated Logout specification] instead.
```
Comment (ahus1) [inline docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc:197]: Fixing a typo, and adding one more sentence to make it hopefully simpler for people to migrate. 

```suggestion
If you were still making use these options or the `redirect_uri` parameter for logout you should implement the link:https://openid.net/specs/openid-connect-rpinitiated-1_0.html[OpenID Connect RP-Initiated Logout specification] instead.
This allows passing in a redirect URL in the `post_logout_redirect_uri` parameter.
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Changes look good to me. Found one typo in the migration guide. Probably worth mentioning this is the release notes as well.

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc (hunk 2 above)
Comment (jonkoops) [reply docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc]: Fixed the typo, as for adding this to the release notes I don't think that is needed. Previous changes to this functionality also have not been added to the release notes, but only the upgrade guide. Adding to much to the release notes just clutters it, and should mostly be reserved for new features.

**Review by ahus1 — DISMISSED** [review]

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc [hunk 3]
```diff
@@ -224,3 +224,14 @@ Update your custom embedded Infinispan cache configuration file with configurati
...
+
+Previous versions of {project_name} had supported automatic logout of the user and redirecting to the application by opening logout endpoint URL such as
+`http(s)://example-host/auth/realms/my-realm-name/protocol/openid-connect/logout?redirect_uri=encodedRedirectUri`. This functionality was deprecated in {project_name} 18 and has been removed in this version in favor of following the OpenID Connect specification.
+
+As part of this change the following related configuration options for the SPI have been removed:
+
+- `--spi-login-protocol-openid-connect-legacy-logout-redirect-uri`
+- `--spi-login-protocol-openid-connect-suppress-logout-confirmation-screen`
+
+If you were still making use these options or the `redirect_uri` parameter for logout you should implement the link:|https://openid.net/specs/openid-connect-rpinitiated-1_0.html[OpenID Connect RP-Initiated Logout specification] instead.
```
Comment (ahus1) [inline docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc:237]:
```suggestion
If you were still making use these options or the `redirect_uri` parameter for logout you should implement the link:https://openid.net/specs/openid-connect-rpinitiated-1_0.html[OpenID Connect RP-Initiated Logout specification] instead.
```

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc (hunk 2 above)
Comment (ahus1) [reply docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc]: You skipped the `|` I removed, added that as well now.

**Review by ahus1 — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc (hunk 2 above)
Comment (jonkoops) [reply docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc]: My bad missed that, thanks

**Pull request merged**